### DISCUSSION TITLE
401 when accessed through an iframe (organizr)

### DISCUSSION THREAD

**Discussion by Krm1t** [authelia/authelia#3896] [Q&A] [answered]
Hello

The discussion at #3429 sounds a bit like my issue but i don't think it's the same.

I have a docker setup that consists of Organizr, SWAG (nginx), Authelia and, some other services.
All my services are accessed through SWAG and almost all services require authentication through Authelia except for Organizr.

The issue i'm having is that if i access any of my services through Organizr then authelia doesn't show the login screen.
Instead it just tries to login with an anonymous user.
If i access the urls directly then the login screen appears as it should.
If i login and then go back to Organizr and hit refresh then it passes authentication and i'm redirected to the service as i should.
I'm assuming that it's because the login is stored in the browser.

I've tried asking on the discord for both Organizr and SWAG and both says that it's definitely not their service that's the problem.
So now i'm here hoping someone has some insights or can tell me what i can do to figure out what the problem is.

The following 2 logs are posted in the container log when i try to access a service through Organizr without first having logged in.
```
level=debug msg="Check authorization of subject username= groups= ip=192.168.1.1 and object https://service.mydomain.com/ (method GET)."
level=info msg="Access to https://service.mydomain.com/ (method GET) is not authorized to user <anonymous>, responding with status code 401" method=GET path=/api/verify remote_ip=192.168.1.1
```

Please let me know if there is any additional information that you need.
Also, this all worked just fine some 3-4 months ago. That is, i didn't update any of my containers for about 3-4 months and then recently i did a full update of all my containers to the latest version.
Unfortunately i was too quick to delete the old versions so i can't tell you which version i was on when it worked.

**Comment by james-d-elliott** [comment, marked as answer]
The only thing that's changed on Authelia's end is that we purposefully block Authelia loading in an `iframe` now as allowing this is fundamentally insecure, if you're unsure why read up about [Clickjacking](https://owasp.org/www-community/attacks/Clickjacking) before taking any further steps.

We will not provide any specific official support allowing this in the future, and will always strongly discourage adjusting these settings and any adjustment is __*AT YOUR OWN RISK*__, however you can purposefully disable this intentional security measure by having the proxy remove the ` X-Frame-Options` response header, and adjust the CSP template here: https://www.authelia.com/configuration/miscellaneous/server/#csp_template to exclude the `frame-ancestors 'none';` option.

**Reply by Krm1t** [reply]
Thanks for the quick answer.
At least now i know why it's not working.
Now i just have to figure out what i wanna do.

**Comment by Krm1t** [comment]
I just thought about something.
Couldn't this problem be mitigated by opening the login page in a new window?
Plex does this when authenticating from 3rd party sites.
